contract/r/gnoswap/pool/v1/pool.gno

```diff
@@ -338,38 +333,55 @@ func collectProtocol(
 	amount0 := u256Min(amount0Req, pool.ProtocolFeesToken0())
 	amount1 := u256Min(amount1Req, pool.ProtocolFeesToken1())
 
-	amount0, amount1 = pool.saveProtocolFees(amount0.Clone(), amount1.Clone())
+	amount0, amount1 = i.saveProtocolFees(pool, amount0.Clone(), amount1.Clone())
 	uAmount0 := safeConvertToInt64(amount0)
 	uAmount1 := safeConvertToInt64(amount1)
 
-	common.SafeGRC20Transfer(cross, pool.token0Path, recipient, uAmount0)
+	common.SafeGRC20Transfer(cross, pool.Token0Path(), recipient, uAmount0)
 	newBalanceToken0, err := updatePoolBalance(pool.BalanceToken0(), pool.BalanceToken1(), amount0, true)
 	if err != nil {
 		panic(err)
 	}
-	pool.balances.token0 = newBalanceToken0
+	pool.SetBalanceToken0(newBalanceToken0)
 
-	common.SafeGRC20Transfer(cross, pool.token1Path, recipient, uAmount1)
+	common.SafeGRC20Transfer(cross, pool.Token1Path(), recipient, uAmount1)
 	newBalanceToken1, err := updatePoolBalance(pool.BalanceToken0(), pool.BalanceToken1(), amount1, false)
 	if err != nil {
 		panic(err)
 	}
-	pool.balances.token1 = newBalanceToken1
+	pool.SetBalanceToken1(newBalanceToken1)
 
 	return amount0.ToString(), amount1.ToString()
 }
 
-func IncreaseObservationCardinalityLimit(
-	cur realm,
+// saveProtocolFees updates the protocol fee balances after collection.
+// Returns amount0, amount1 representing the fees deducted from protocol reserves.
+func (i *poolV1) saveProtocolFees(pool *pl.Pool, amount0, amount1 *u256.Uint) (*u256.Uint, *u256.Uint) {
+	token0Fees, underflow := u256.Zero().SubOverflow(pool.ProtocolFeesToken0(), amount0)
+	if underflow {
+		panic(errOverFlow)
+	}
+	pool.SetProtocolFeesToken0(token0Fees)
+
+	token1Fees, underflow := u256.Zero().SubOverflow(pool.ProtocolFeesToken1(), amount1)
+	if underflow {
+		panic(errOverFlow)
+	}
```

Comment:
error message and variable name's are do not matched here. I think an underflow error should be returned.